observer/probers/http/http.go

```diff
@@ -60,5 +63,6 @@ func (p HTTPProbe) Probe(timeout time.Duration) (bool, time.Duration) {
 	if err != nil {
 		return false, time.Since(start)
 	}
+	p.notAfter.WithLabelValues(p.url).Set(float64(resp.TLS.PeerCertificates[0].NotAfter.Unix()))
```

Comment:
What if the fetched URL truly is HTTP, and doesn't have a chain of certificates at all? Then at the very least `PeerCertificates[0]` will be an index-out-of-bounds error, and maybe even `resp.TLS` will be a nil pointer exception.

Reply:
Would it be bad practice to instead export zero if the URL is HTTP and export the expiration date otherwise? Most of the use cases where I want the expiration are HTTPS, and there are workaround for the cases where I'm using HTTP.

Reply:
I guess my fundamental question is "why have the expiration date associated with the HTTP prober at all, rather than the TLS prober?". I think of the HTTP prober as being "does the site load and give a 200" while the TLS prober is "what certs does the site serve, regardless of response code" -- since this cares about the certs, I'd imagine it associated with the TLS prober.

Is there a difficulty to configuring a bunch of additional TLS probers for the sites that we care about cert expiration for?

Reply:
The reason I was initially avoiding using TLS prober also checks that the root cert of the website is the expected one and I didn't want to go and find the root cert of each site I wanted to do the check for.  But now that you mention it, what if instead of reconfiguring the HTTP prober, I reconfigure the TLS prober so that it only checks the root if the settings provide a root?

Reply:
Ooh yeah I like that idea.

Reply:
Awesome, thanks! I opened a pull request at https://github.com/letsencrypt/boulder/pull/6569 and I'll close this one.